log/src/main/java/kcl/teamIndexZero/traffic/log/Logger.java

```diff
@@ -26,15 +32,15 @@ private Logger(String instance_name) {
      * @param instance_name Name of the Logger instance
      * @return Logger
      */
-    public static Logger getLoggerInstance(String instance_name) {
-        if (loggers == null) loggers = new HashMap<String, Logger>();
-        if (log_engine == null) log_engine = Log_Engine.getInstance();
-        Logger logger_instance = loggers.get(instance_name);
-        if (logger_instance == null) {
-            logger_instance = new Logger(instance_name);
-            loggers.put(instance_name, logger_instance);
+    public static synchronized Logger getLoggerInstance(String instance_name) {
+        synchronized (Logger.class) {
```

Comment:
I would guess that both `synchronized` objects here are using the same synchronization monitor - the one belonging to Logger.class. It does not cause too much problem since the locks are luckily reenterant, however seems a bit an overkill.